test/feature/test_hardnet.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+from test.common import device
+
+import torch
+from torch.testing import assert_allclose
+from torch.autograd import gradcheck
+
+from kornia.feature.hardnet import HardNet
+import kornia.testing as utils  # test utils
+
+
+class TestHardNet:
+    def test_shape(self, device):
+        inp = torch.ones(1, 1, 32, 32, device=device)
+        hardnet = HardNet().to(device)
+        hardnet.eval()  # batchnorm with size 1 is not allowed in train mode
+        out = hardnet(inp)
+        assert out.shape == (1, 128)
+
+    def test_shape_batch(self, device):
+        inp = torch.ones(16, 1, 32, 32, device=device)
+        hardnet = HardNet().to(device)
+        out = hardnet(inp)
+        assert out.shape == (16, 128)
+
+    @pytest.mark.skip("jacobian not well computed")
```

Comment:
@ducha-aiki any idea why would that happen ?

Reply:
What, why liberty aug is the best? No idea except may be that the rest of datasets are simpler.

Reply:
nope, gradcheck failing

Reply:
detach in input norm, I guess.